### ISSUE TITLE
Get the project ID based on the name

### ISSUE THREAD

**Issue by russellseymour** [Infisical/infisical#2594]
### Feature description
When working with the API, it requires the ID of the project to work with.

This is a GUID so is not easy to remember. So this request to add an endpoint that returns the ID of a valid project based on the name.

### Why would it be useful?
When wokring with the API, people will remember the name of the project over the ID. I am looking to build an automated system that interacts with Infisical and I need to be able to allow the project name to be specified, however I then need to translate this into the project id,

### Additional context
The only way I can see to get the ID at the moment is to look at the UI and see the ID in the URL. This means people have to log into the UI to get the ID before they can run the automated process using the API

**Comment by shreyas4510** [comment]
Please assign this issue to me

**Comment by akhilmhdh** [comment]
Agreed. This what we are also planning to do. All endpoints will move to id based and then there will slug to id converter endpoints. 

For now you can use project list api to filter out and find the id but the endpoint for project will be published soon

**Comment by mdayanabbas** [comment]
Hey try out these approches once:
Make sure Bun is installed and accessible (use which bun).
Try infisical run -- /path/to/bun run index.ts
Set environment variable PATH to include Bun's location.
Create a script to set up Bun and run it with Infisical.
Look for specific Bun instructions or known issues.

**Comment by guptave2** [comment]
Hi, I apologize for not asking to be assigned before submitting the PR. I've already made the changes and submitted [PR #2714]. Please let me know if there's anything else you'd like me to do!

**Comment by akhilmhdh** [comment]
Closing as completed

Reference: https://infisical.com/docs/api-reference/endpoints/projects/get-project-by-slug
